src/v2/Apps/Auction/Routes/Register/index.tsx

```diff
@@ -97,33 +108,35 @@ export const RegisterRoute: React.FC<RegisterProps> = props => {
     }
 
     trackRegistrationFailed(errorMessages)
-
-    actions.setSubmitting(false)
     actions.setStatus("submissionFailed")
   }
 
-  function handleSubmit(actions: FormikActions<object>, result: FormResult) {
-    createCreditCardAndUpdatePhone(
-      relay.environment,
-      result.phoneNumber,
-      result.token.id
-    )
-      .then(() => {
-        createBidder()
-          .then((data: RegisterCreateBidderMutationResponse) => {
-            trackRegistrationSuccess(data.createBidder.bidder.internalID)
-
-            window.location.assign(
-              `${sd.APP_URL}/auction/${sale.slug}/confirm-registration`
-            )
-          })
-          .catch(error => {
-            handleMutationError(actions, error)
-          })
-      })
-      .catch(error => {
-        handleMutationError(actions, error)
-      })
+  async function createTokenAndSubmit(
```

Comment:
for my own education – what's the thinking between having these function be declared within the "namespace" of the component that is using them (before this commit) vs just injected the functions into the component, but defining them outside of the component "namespace" as you are now currently doing?

Any particulars of JS/TS to consider? Or is there a general code organization point?

Reply:
* It creates more refactoring opportunities. For instance, the `CreateCreditCardAndUpdatePhone` mutation was [part of the function component](https://github.com/artsy/reaction/blob/549f190198f16321694cc096d680b6de674957ff/src/Apps/Auction/Routes/Register/index.tsx#L91) and then [extracted](https://github.com/artsy/reaction/commit/e2d6154ba8bdba8c04ec56b06d7a7d0ca2652b7c#diff-db7fa0c494cb2467a50b4b0bbc55391fR40) out of it, and now it sits in [its own file](CreateCreditCardAndUpdatePhone) shared by two different components. Now there is another refactoring opportunity for the `saleConfirmRegistrationPath` function and [the `confirmRegistrationPath ` function](https://github.com/artsy/force/blob/a190b8ddf4d4adc473f6b868b11e12658e78fb0b/src/v2/Apps/Auction/getRedirect.ts#L79-L80).
 * This is micro optimization, but functions within the component get re-defined every time it gets re-rendered. That makes [the JIT compiler](https://v8.dev/blog/jitless) harder to optimize the runtime performance.